webui/react/src/stores/experiments.tsx

```diff
@@ -4,93 +4,74 @@ import { getExperiments } from 'services/api';
 import { V1Pagination } from 'services/api-ts-sdk';
 import { GetExperimentsParams } from 'services/types';
 import { ExperimentItem, ExperimentPagination } from 'types';
-import handleError from 'utils/error';
 import { Loadable, Loaded, NotLoaded } from 'utils/loadable';
 import { Observable, observable, WritableObservable } from 'utils/observable';
 import { encodeParams } from 'utils/store';
 
-type ExperimentsCache = {
+import PollingStore from './polling';
+
+type ExperimentCache = {
   experimentIds: Readonly<number[]>;
   pagination: Readonly<V1Pagination>;
 };
 
-// TODO: use js private intead of ts private for methods
-class ExperimentsService {
-  #experimentsCache: WritableObservable<Map<string, ExperimentsCache>> = observable(Map());
+class ExperimentStore extends PollingStore {
+  // Cache values keyed by encoded request param.
+  #experimentCache: WritableObservable<Map<string, ExperimentCache>> = observable(Map());
   #experimentMap: WritableObservable<Map<number, ExperimentItem>> = observable(Map());
 
-  // Get an experiment by experiment id
-  public getExperimentsByIds(experimentIds: number[]): Observable<Loadable<ExperimentItem[]>> {
-    return this.#experimentMap.select((map) => {
-      const expList: ExperimentItem[] = experimentIds.flatMap((id) => {
+  public getExperimentsByIds(experimentIds: number[]): Observable<ExperimentItem[]> {
+    return this.#experimentMap.select((map) =>
+      experimentIds.flatMap((id) => {
         const exp = map.get(id);
         return exp ? [exp] : [];
-      });
-      return Loaded(expList);
-    });
+      }),
+    );
   }
 
   public getExperimentsByParams(
-    params: Readonly<GetExperimentsParams>,
+    params: GetExperimentsParams,
```

Comment:
question: why is `Readonly` removed?

Reply:
I guess it was a combination of the functions not really modifying the params and just in general trying to have functions not mutate the input params. Though maybe this is a good extra protection, is the idea to add `Readonly<T>` everywhere it is true?

Reply:
even if its obvious that the function doesn't modify the params, i think its always nice to add `readonly`. Especially if function gets long, its hard to tell that params are modified or not. that requires additional debugging to understand the code in the future without `readonly`. 
i think its good for us to add Readonly<T> everywhere. I actively add it when i noticed it

Reply:
Also an _especially_ good idea to add it to the return types.